solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -801,7 +815,11 @@ public void load() {
     ClusterAPI clusterAPI = new ClusterAPI(collectionsHandler, configSetsHandler);
     containerHandlers.getApiBag().registerObject(clusterAPI);
     containerHandlers.getApiBag().registerObject(clusterAPI.commands);
-    containerHandlers.getApiBag().registerObject(clusterAPI.configSetCommands);
+    containerHandlers.getApiBag().registerObject(new CreateConfigSetAPI(this));
+    containerHandlers.getApiBag().registerObject(new DeleteConfigSetAPI(this));
+    containerHandlers.getApiBag().registerObject(new ListConfigSetsAPI(this));
+    containerHandlers.getApiBag().registerObject(new UploadConfigSetAPI(this));
+    containerHandlers.getApiBag().registerObject(new UploadConfigSetFileAPI(this));
```

Comment:
This is fine for now, but I think we're going to want a better approach for it, as we convert more APIs I suspect the list will get long and unwieldy and easy to miss something or make mistakes. Maybe an annotation based discovery framework, hopefully that won't make startup time too long.

Reply:
Yeah, it's not ideal for sure.  Core-level APIs are registered out of solrconfig.xml's, so they don't have the same downside.  It's just the Solr/container-level APIs.  But still, it won't scale out as we split up more APIs.

I did create an ApiRegistrar class to group things together, so that CoreContainer can register whole groups of APIs with a single call.  So maybe I'll go that route with these APIs in the short term.  I should've done that out of the gate, my mistake.

As for an annotation-discovery framework - I'd love to use an existing one.  But IMO there are too many good options out there to justify writing our own.  I've proposed JAX-RS in the past as a fix for a wider set of problems but the idea had lukewarm reception, but maybe opinion will have changed on that, or maybe there's some lighter alternative to solve this problem only.

Reply:
I've moved these out to ApiRegistrar, which at least gives us some short-term improvement here.